src/coreclr/vm/debugdebugger.cpp

```diff
@@ -532,182 +535,208 @@ FCIMPL4(void, DebugStackTrace::GetStackFramesInternal,
+                                                documents [x]->Release();
+                                            }
+                                        } // if got sequence points
 
-                                }  // if all memory allocations succeeded
+                                    }  // if all memory allocations succeeded
 
-                                // holders will now delete the arrays.
+                                    // holders will now delete the arrays.
+                                }
                             }
+                            // Holder will release pISymUnmanagedMethod
                         }
-                        // Holder will release pISymUnmanagedMethod
-                    }
 
-                } // GCX_PREEMP()
+                    } // GCX_PREEMP()
 
-                if (fFileInfoSet)
-                {
-                    // Set the line and column numbers
-                    I4 *pI4Line = (I4 *)((I4ARRAYREF)pStackFrameHelper->rgiLineNumber)->GetDirectPointerToNonObjectElements();
-                    pI4Line[iNumValidFrames] = sourceLine;
+                    if (fFileInfoSet)
+                    {
+                        // Set the line and column numbers
+                        I4 *pI4Line = (I4 *)((I4ARRAYREF)pStackFrameHelper->rgiLineNumber)->GetDirectPointerToNonObjectElements();
+                        pI4Line[iNumValidFrames] = sourceLine;
 
-                    I4 *pI4Column = (I4 *)((I4ARRAYREF)pStackFrameHelper->rgiColumnNumber)->GetDirectPointerToNonObjectElements();
-                    pI4Column[iNumValidFrames] = sourceColumn;
+                        I4 *pI4Column = (I4 *)((I4ARRAYREF)pStackFrameHelper->rgiColumnNumber)->GetDirectPointerToNonObjectElements();
+                        pI4Column[iNumValidFrames] = sourceColumn;
 
-                    // Set the file name
-                    OBJECTREF obj = (OBJECTREF) StringObject::NewString(wszFileName);
-                    pStackFrameHelper->rgFilename->SetAt(iNumValidFrames, obj);
+                        // Set the file name
+                        OBJECTREF obj = (OBJECTREF) StringObject::NewString(wszFileName);
+                        pStackFrameHelper->rgFilename->SetAt(iNumValidFrames, obj);
+                    }
                 }
-#endif // FEATURE_ISYM_READER
 
-                // If the above isym reader code did NOT set the source info either because it is ifdef'ed out (on xplat)
-                // or because the pdb is the new portable format on Windows then set the information needed to call the
-                // portable pdb reader in the StackTraceHelper.
+                // If the above isym reader code did NOT set the source info either because the pdb is the new portable format on
+                // Windows then set the information needed to call the portable pdb reader in the StackTraceHelper.
                 if (fPortablePDB)
+#endif // FEATURE_ISYM_READER
```

Comment:
~If FEATURE_ISYM_READER is not defined, does that mean that we will always go down this path and do all the work?~ I see this is the existing behavior. Do you know why we don't use the same codeview iteration logic here?